### ISSUE TITLE
[NEWS] Anulowanie zamówień przez kupującego / Orders cancellation by the buyer

### ISSUE THREAD

**Issue by MaciejFrackowiak** [allegro/allegro-api#3085]
Od **05.05.2020** kupujący będą mogli anulować zamówienie w ciągu 3 dni od daty zakupu. Udostępnimy taką opcję, gdy:

- sprzedawca posiada konto Firma;
- status realizacji zamówienia (“fulfillment.status”) jest inny niż “PROCESSING”, “READY_FOR_SHIPMENT”, “SENT”;
- sprzedawca nie dodał numeru przesyłki do zamówienia.

 W związku z tym dla zasobów:

- [GET /order/events](https://developer.allegro.pl/documentation/#operation/getOrderEventsUsingGET) - dodamy nowy typ eventu **“BUYER_CANCELLED”**. Jeżeli zobaczysz w dzienniku zdarzeń taki wpis, oznacza to, że kupujący zrezygnował z realizacji zamówienia. 

     **Ważne!** Zdarzenie “BUYER_CANCELLED” może pojawić się przed lub po READY_FOR_PROCESSING.

- [GET /order/checkout-forms](https://developer.allegro.pl/documentation/#operation/getListOfOrdersUsingGET) oraz [GET /order/checkout-forms/{checkoutFormId}](https://developer.allegro.pl/documentation/#operation/getOrdersDetailsUsingGET) - dodamy nowy status zamówienia **“CANCELLED”**. 

Jeżeli kupujący anulował zamówienie już opłacone, możesz zwrócić otrzymaną wpłatę za pomocą [POST /payments/refunds](https://developer.allegro.pl/documentation/#operation/initiateRefund). Dla niezrealizowanych zamówień możesz wystąpić o rabat transakcyjny przy użyciu [POST /order/refund-claims](https://developer.allegro.pl/documentation/#operation/createRefundApplication).

Więcej informacji o anulowaniu zamówień znajdziesz na stronie [dla sprzedających](https://allegro.pl/dla-sprzedajacych/juz-w-maju-kupujacy-anuluja-zamowienie-w-nowy-sposob-rKDg19Gkmh3).

-----

**On May 5, 2020** we will introduce to buyers option to cancel an order within 3 days after purchase. Buyer will be able to cancel an order if:

- seller has a company account;
- order fulfillment status (“fulfillment.status”) is different than PROCESSING, READY_FOR_SHIPMENT, SENT;
- seller has not added tracking number to the order.

Therefore, to the following resources:

- [GET /order/events](https://developer.allegro.pl/documentation/#operation/getOrderEventsUsingGET) - we will add a new event type **“BUYER_CANCELLED”**. If you see such entry in the event log, it means that the buyer cancelled the order.

    **Note!** Event “BUYER_CANCELLED” may appear before or after     “READY_FOR_PROCESING”.

- [GET /order/checkout-forms](https://developer.allegro.pl/documentation/#operation/getListOfOrdersUsingGET) and [GET /order/checkout-forms/{checkoutFormId}](https://developer.allegro.pl/documentation/#operation/getOrdersDetailsUsingGET) - we will add a new order status **“CANCELLED”**. 

If the buyer cancelled an order, which is already paid, you can refund the payment via [POST /payments/refunds](https://developer.allegro.pl/documentation/#operation/initiateRefund). For cancelled orders you can create a sale commission refund application via [POST /order/refund-claims](https://developer.allegro.pl/documentation/#operation/createRefundApplication).

You can find more information about orders cancellation on the [for sellers page](https://allegro.pl/dla-sprzedajacych/juz-w-maju-kupujacy-anuluja-zamowienie-w-nowy-sposob-rKDg19Gkmh3).

**Comment by pawelosajda** [comment]
Czy możliwość anulowania zamówienia jest już dostępna w środowisku testowym?

**Comment by MaciejFrackowiak** [comment]
Nie, ponieważ trwają jeszcze ostatnie prace nad produkcyjną wersją mechanizmu.

**Comment by pietrach** [comment]
Ale jako sprzedający też mogę zmienić status zamówienia na CANCELLED ? Jednocześnie też mogę mieć taki status przy pobraniu zamówienia i to oznacza, że klient anulował ?

**Comment by MaciejFrackowiak** [comment]
@pietrach Nie, status zamówienia na CANCELLED (dla GET /order/checkout-forms i GET /order/checkout-forms/{checkoutFormId}) zmieni się tylko pod wpływem działania kupującego. Natomiast Ty ze swojej strony możesz zmienić status realizacj każdego zamówienia (fulfillment.status) korzystając z [PUT /order/checkout-forms/{id}/fulfillment](https://developer.allegro.pl/documentation/#operation/setOrderFulfillmentUsingPUT) na CANCELLED.

**Comment by pietrach** [comment]
To w takim razie jaki będzie status zamówienia jeśli zmienię poprzez API  PUT /order/checkout-forms/{id}/fulfillment na CANCELLED i później pobiorę jego dane GET /order/checkout-forms i GET /order/checkout-forms/{checkoutFormId} ?

**Comment by MaciejFrackowiak** [comment]
@pietrach Musisz wziąć pod uwagę, że obecnie sprzedający będzie miał możliwość czasowego wyłączenia opcji anulowania zamówienia przez kupujących - i tu pojawiają się dwie ścieżki:

1) sprzedający wyłączył opcję --> status zamówienia nigdy nie zmieni się na CANCELLED (ostatni możliwy to RFP), stąd Ty jako sprzedący zmieniasz tylko fulfillment.status;

2) sprzedający ma aktywną opcję anulowania --> status zamówienia automatycznie (na skutek działania kupującego) zmieni się na CANCELLED i żadna Twoja operacja nie zmieni statusu zamówienia, zmienisz tylko fulfillment.status.

**Comment by Onixarts** [comment]
Czy jeśli kupujący anuluje zamówienie, to czy przy próbie ustawienia statusu zamówienia przez sprzdawcę np na PROCESSING albo READY_FOR_SHIPMENT zasób `UT /order/checkout-forms/{id}/fulfillment ` zwróci jakiś błąd?
Jeśli nie, to sprzedawca nie dowie się o tym, że klient anulował w tym samym czasie zamówienie dopóki nie pobierze dziennika zdarzeń albo danych zamówienia.

**Comment by pietrach** [comment]
@MaciejFrackowiak ale ja na chwilę obecną mogę wysłać PUT /order/checkout-forms/{id}/fulfillment na CANCELLED i zamówienia jest anulowane przez sprzedającego. I wtedy jako fulfillment.status pojawi mi się "CANCELLED" ?
Co masz na myśli "Musisz wziąć pod uwagę, że obecnie sprzedający będzie miał możliwość czasowego wyłączenia opcji anulowania zamówienia przez kupujących" ?
Że wprowadzacie taką opcję ale będzie można z niej nie korzystać wogóle ?

**Comment by MaciejFrackowiak** [comment]
@Onixarts Nie, statusy realizacji zamówienia (fulfillment.status) zależą tylko od działania sprzedającego - może zawsze zmienić jego wartość na dowolną. Tak, aby sprawdzić czy kupujący anulował zamówienie niezbędne jest sprawdzenie dziennika zdarzeń lub bezpośrednio checkoutForms.

**Comment by MaciejFrackowiak** [comment]
@pietrach Tak, to będzie tylko zmiana porządkowa dokonana przez sprzedającego i będzie miała odniesienie wyłącznie do fulfillment.status.
W podlinkowanym przeze mnie artykule dołączone jest szczegółowe info:

> Aby każdy sprzedający mógł przygotować się do nowego sposobu anulowania zamówienia przez kupującego, do końca lipca obowiązywać będzie okres przejściowy. W tym czasie możesz wyłączyć opcję anulowania zakupu w swoich ofertach.

>

**Comment by Onixarts** [comment]
To trochę słabo. Teraz przed każdym ruchem sprzedawca będzie musiał pobierać dane z Allegro i całe pobieranie danych i trzymanie we własnej bazie staje się bez sensu, skoro wymuszacie w ten sposób pracę online.

**Comment by pietrach** [comment]
@MaciejFrackowiak w takim razie po jakich danych w  API rozpoznać, czy anulował sprzedawca czy kupujący ?

**Comment by MaciejFrackowiak** [comment]
@Onixarts Tylko opierając się na dzienniku zdarzeń otrzymasz (najszybciej) na bieżąco informację, czy kupujący anulował zamówienie.

**Comment by MaciejFrackowiak** [comment]
@pietrach Jeżeli kupujący anulował zamówienie to ujrzysz event BUYER_CANCELLED oraz checkoutForm otrzyma status CANCELLED. Zmiany sprzedającego widoczne są tylko w statusie realizacji zamówienia (fulfillment.status).

**Comment by pietrach** [comment]
@MaciejFrackowiak ale ja nie korzystam z eventów wogóle. Czy nie może być takiej informacji tutaj GET /order/checkout-forms/{checkoutFormId} ?

**Comment by MaciejFrackowiak** [comment]
@pietrach Ale oczywiście też będzie - spójrz proszę do treści komunikatu.

**Comment by pietrach** [comment]
Czyli dodacie do pola status dodatkową wartość "BUYER_CANCELLED" lub "CANCELLED" ?
Teraz możliwe wartości to READY_FOR_PROCESSING, FILLED_IN, BOUGHT, FULFILLMENT_STATUS_CHANGED .

**Comment by MaciejFrackowiak** [comment]
@pietrach Dokładnie tak:

> GET /order/checkout-forms oraz GET /order/checkout-forms/{checkoutFormId} - dodamy nowy status zamówienia “CANCELLED”.

>

**Comment by Onixarts** [comment]
@MaciejFrackowiak 
>Tylko opierając się na dzienniku zdarzeń otrzymasz (najszybciej) na bieżąco informację, czy kupujący anulował zamówienie.

To ja wiem, ale nie do końca czujecie chyba jak działają programy korzystające z API. Nie ciągną one przecież na żywo non stop dziennika zdarzeń, tylko robią to co jakiś czas. Nawet niech to będzie minuta to i tak w tej minucie kupujący może anulować zamówienie a sprzedawca lub jego system zacznie je przetwarzać i nie ma żadnej informacji o tym, że jest konflikt. W systemie sprzedawcy wygenerują się dokumenty, faktury, zostaną zafiskalizowane i trzeba to potem odkręcać. 
Aby zrobić to po waszemu, przed każdą operacją wykonaną w systemie sprzedawcy trzeba by ciągnąć dane zamówienia z Allegro, a pamiętajcie że tych zamówień niektórzy mają więcej niż 1 na godzinę i pobranie tych danych oraz samo ich przetwarzanie trwa pewien czas w którym w serwisie kupujący może anulować. Programiści i testerzy powinni u was pracować na większym zbiorcze danych to zobaczyliby w czym problem i że czas wymiany i przetwarzania danych to nie są milisekundy a często kilkadziesiąt sekund albo i minut.
Moim zdaniem ustawienie przez sprzedawcę statusu wskazującego na przetwarzanie powinno generować błąd jeśli kupujący anulował zamówienie - inaczej będzie patologia, niby anulowane a jednak mogę sobie oznaczyć, że wysłane.

**Comment by MaciejFrackowiak** [comment]
@Onixarts Dziękuję za przedstawienie przez Ciebie spostrzeżeń. Obraliśmy natomiast kierunek, w którym zmiany kupującego widoczne są w eventach i checkoutFormie, natomiast sprzedającego w statusie realizacji zamówienia - są to dwie niezależne ścieżki, co pozwala jednoznacznie odpowiedzieć kto podjął działanie.

**Comment by MaciejFrackowiak** [comment]
@Onixarts W opisanym przez Ciebie przypadku wystarczy, że sprzedający w momencie rozpoczęcia przetwarzania zamówienia (a zakładam, że robi to dopiero, gdy status to READY_FOR_PROCESIING) dokona zmiany statusu realizacji zamówienia (fulfillment.status) na np. "PROCESSING" (w realizacji) i już kupujący nie może anulować zamówienia. W tym momencie nie dojdzie do "pomieszania".

**Comment by SebastianOzdoba** [comment]
@MaciejFrackowiak załóżmy jednak że może zdarzyć się sytuacja w której prawie jednocześnie ściągniemy zamówienie i Klient oznaczy CANCELLED lub wystąpi błąd w restAPI (dzienniki zareagują z opóźnieniem, co się zdarza)  / aplikacji integrującej
Co się stanie jeśli dla zamówienia CANCELLED zostanie wysłane PROCESSING
Który z systemów jest nadrzędny i czy w odpowiedzi zostanie zwrócona jasna informacja że jest anulowane, czy dostaniemy cokolwiek co pozwoli nam przerwać proces 
Czytam powyższe posty ale nie ma jasno opisanego procesu i znów z tego wyjdą nieporozumienia

**Comment by pietrach** [comment]
@MaciejFrackowiak czy na chwilę obecną przed 5 maja jak użyje filtrów w metodzie `GET /order/checkout-forms` => `status=CANCELLED` to metoda wyrzuci błąd ? Bo testuje sobie z filtrami `lineItems.boughtAt.gte, lineItems.boughtAt.lte, status, offset, limit` i wyrzuca mi błąd:

`{"code":"ERROR","message":"An error has occurred","details":null,"path":"\/order\/checkout-forms","userMessage":null}`

**Comment by MaciejFrackowiak** [comment]
@SebastianOzdoba odpowiadając na:

>  Co się stanie jeśli dla zamówienia CANCELLED zostanie wysłane PROCESSING

W tej sytuacji PROCESSING będzie tylko porządkowym stanem realizacji zamówienia przez sprzedającego (nie wpłynie w żaden sposób na zamówienie, które jest już anulowane przez kupującego.)

Jak wspomniałem we wcześniejszym poście:

> Obraliśmy natomiast kierunek, w którym zmiany kupującego widoczne są w eventach i checkoutFormie, natomiast sprzedającego w statusie realizacji zamówienia - są to dwie niezależne ścieżki, co pozwala jednoznacznie odpowiedzieć kto podjął działanie.

**Comment by MaciejFrackowiak** [comment]
@pietrach Wszelkie zmiany produkcyjnie pojawią się 5 maja.

**Comment by SebastianOzdoba** [comment]
@MaciejFrackowiak ale to bez sensu jaki "porządkowy" stan. Stan zamówienia jest jednoznaczny albo jest anulowane albo jest procesowane. Więc przy próbie wysłania PROCESSING dla CANCELLED powinien być zwrócony błąd

**Comment by Onixarts** [comment]
@MaciejFrackowiak 

> W opisanym przez Ciebie przypadku wystarczy, że sprzedający w momencie rozpoczęcia przetwarzania zamówienia (a zakładam, że robi to dopiero, gdy status to READY_FOR_PROCESIING) dokona zmiany statusu realizacji zamówienia (fulfillment.status) na np. "PROCESSING

No właśnie nie wystarczy. To byłby świat idealny. Sprzedawca pobiera zamówienia w poniedziałek rano z weekendu. Trwa to np 2 minuty, bo jest ich sporo. Przechodzi do realizacji pierwszego z nich, status RFP. System wysyła do API status PROCESSING i zadowolony sprzedawca kontynuuje obsługę zamówienia. Nie wie jednak, że chwilę wcześniej kupujący anulował zamówienie. Dowie się dopiero, jak system ponownie sięgnie po dziennik zdarzeń, a robi to co pewien czas. I tutaj dochodzi do sytuacji wyścigu dwóch procesów. Tylko, że sprzedawca ma wpływ na zablokowanie możliwości anulowania jak będzie pierwszy, zaś jak kupujący będzie pierwszy to dochodzi do rozjazdu statusów.

**Comment by MaciejFrackowiak** [comment]
@Onixarts @SebastianOzdoba @pietrach Przeanalizowaliśmy Wasze spostrzeżenia i chcemy zaproponować koncepcję w podejściu do anulowania zamówień, którą szczegółowo opisaliśmy w [dokumencie](https://github.com/allegro/allegro-api/files/4528637/Pole.revision.w.Zamowieniach.-.koncepcja.pdf). Proszę, zapoznajcie się z nią - liczymy na Wasze komentarze.

**Comment by Onixarts** [comment]
Jak dla mnie wygląda całkiem w porządku i rozwiązuje problemy. Jestem na tak.

**Comment by TimothyKoval** [comment]
Wszystko fajnie, ktoś w końcu pomyślał dla odmiany. Tylko w takim przypadku przesuńcie termin z 5 maja, bo to są większe zmiany w systemie zamówień, a jest to elementarny element integracji w każdej platformie sprzedażowej.

**Comment by SebastianOzdoba** [comment]
Dla nas też to rozwiązanie jest dużo lepsze, z tym że termin 5maja mało realny by to wdrożyć

**Comment by pietrach** [comment]
@MaciejFrackowiak OK

**Comment by pietrach** [comment]
@MaciejFrackowiak a co np. jeśli przy pobieraniu zamówień zmienimy im status na w realizacji a ich rzeczywisty czas realizacji będzie powiedzmy za 2-3 dni. Chodzi o to aby się zabezpieczyć przed anulacją przez kupującego. To dobry pomysł czy Allegro sprawdza ile czasu się realizuje zamówienie i później ma to wpływ na jakieś oceny itp. ?

**Comment by Onixarts** [comment]
Hehe, już knujecie funkcję 
- [x]  oznacz nowe zamówienia jako w trakcie po odebraniu? ;)
Też o niej przez chwilę pomyślałem, bo zapewne nasi klienci o to wkrótce zapytają.

**Comment by MaciejFrackowiak** [comment]
@Onixarts @SebastianOzdoba @pietrach @TimothyKoval Dziękuję za Wasze opinie nt. proponowanej zmiany. Finalnie, wdrożymy rozwiązanie z "revision", jednak nie wpłynie to na datę wprowadzenia opcji anulowania zamówień w Allegro. Każdy sprzedający będzie mógł wyłączyć do lipca opcję w zakładce [Ustawienia zamówień](https://allegro.pl/moje-allegro/sprzedaz/zamowienia/ustawienia).

**Comment by MaciejFrackowiak** [comment]
@pietrach W chwili obecnej taka zmiana statusów nie wpływa na oceny.

**Comment by pietrach** [comment]
@MaciejFrackowiak a czy może wpłynąć na oceny za jakiś czas ?

Aby to wyłączyć to trzeba mieć abonament conajmniej podstawowy wykupiony czy niekoniecznie ?

**Comment by Onixarts** [comment]
A kiedy to będzie działać na sandboxie?

**Comment by MaciejFrackowiak** [comment]
@pietrach Nie, opcja zostanie udostępniona dla każdego sprzedającego, nie trzeba będzie posiadać abonamentu.

@Onixarts Wraz z wejściem na produkcję funkcjonalność zostanie dodana na Sandbox.

**Comment by SebastianOzdoba** [comment]
@MaciejFrackowiak kiedy będzie dostępny nowy plik  YAML? Obecny nie ma danych dotyczących tego procesu

**Comment by MaciejFrackowiak** [comment]
@SebastianOzdoba Zmiana planowo wdrożona zostanie 5 maja 2020 i wówczas pojawi się aktualna wersja swagger.yaml.

**Comment by SebastianOzdoba** [comment]
@MaciejFrackowiak prośba na przyszłość by publikować takie rzeczy wcześniej, tak samo jak i zmiany na sandbox. Nawet nie możemy przetestować ani nawet zacząć implementacji (serializujemy sobie yaml) przed umieszczeniem przez Was na produkcji :)

**Comment by alekskuc** [comment]
Witam, @MaciejFrackowiak chciałbym zaimplementować zmiany odnośnie anulowania zamówień "na sucho".
Czy to są wszystkie przewidywane zmiany w dokumentacji swagger.yaml? 
Proszę o weryfikację.

![image](https://user-images.githubusercontent.com/8465007/80710733-87e57480-8aef-11ea-9cb2-5c89a2971674.png)

![image](https://user-images.githubusercontent.com/8465007/80710789-9c297180-8aef-11ea-8524-93999b23587e.png)

**Comment by Onixarts** [comment]
@MaciejFrackowiak czy 5 maja będzie również zaimplementowane rozwiązanie z revision opisane w dokumencie? Czy to pojawi się później?

**Comment by MaciejFrackowiak** [comment]
@Onixarts Pole "revision" pojawi się później, dlatego każdy sprzedający dostanie możliwość wyłączenia opcji anulowania zamówień (do lipca) w zakładce [Ustawienia zamówień](https://allegro.pl/moje-allegro/sprzedaz/zamowienia/ustawienia).

**Comment by pietrach** [comment]
Czy to już zostało wdrożone ?
Bo szukam w zakładce ustawienia dla zamówień opcji wyłączenia tego i nie ma.

**Comment by MaciejFrackowiak** [comment]
@pietrach Opcja w [ustawieniach](https://allegro.pl/moje-allegro/sprzedaz/zamowienia/ustawienia) jest już widoczna.

**Comment by SebastianOzdoba** [comment]
@MaciejFrackowiak jak mogę anulować zamówienie na sandbox? Widzę że są nowe ustawienia na koncie firmowym, ale jako kupujacy w szczegółach zamówienia nie widzę opcji anuluj
Kupuję, opłacam, wchodzę na konto do szczegółów zamówienia i nic nie ma :) Może jestem ślepy :)

**Comment by MaciejFrackowiak** [comment]
@SebastianOzdoba Opcja dostępna jest w szczegółach Twojego zakupu:

![obraz](https://user-images.githubusercontent.com/53566763/81292231-81b14400-906b-11ea-9bca-eed451b657bc.png)

**Comment by SebastianOzdoba** [comment]
Dobra ;), ktoś robił mi dywersję :)

**Comment by pietrach** [comment]
To pytanie chyba już tu zadałem ale nie uzyskałem na nie ostatecznej odpowiedzi. 
Planuje zrobić tak, że podczas pobierania ofert zmienię ofercie status na "w realizacji". Pytanie czy Allegro w przyszłości nie zamierza badać jak długo jest zamówienie w danym statusie i wyciągać z tego jakieś konsekwencje itp. ? Chodzi o to, że załóżmy klient złoży ofertę w Boże Narodzenie. Zostanie ona od razu pobrana i zmieniony status na "w realizacji". Po czym to zamówienie w rzeczywistości zacznie być realizowane w Nowym Roku itp. Czy takie ustawianie statusu z wyprzedzeniem tylko po to aby zapobiec blokadzie anulowania oferty przez klienta jest dobrym rozwiązaniem ? Nie wpłynie to na jakieś oceny przez Allegro i ewentualnie klientów w przyszłości ?

**Comment by MaciejFrackowiak** [comment]
@pietrach Mogę Tobie odpowiedzieć na to pytanie zakładając stan obecny - nie, nie jest na tej podstawie brany jakikolwiek czynnik do oceny jakości sprzedaży, etc. Natomiast co się wydarzy w przyszłości to trudno na ten moment prorokować - na pewno jeżeli w jakikolwiek sposób będzie to brane pod uwagę, to zostanie to zakomunikowane.

**Comment by magik092** [comment]
@MaciejFrackowiak na sandbox nie widzę opcji anulowania pomimo, że konto mam niby jako firma oraz w zakładce zamówień mam włączoną opcję anulowanych zamówień.
Coś może być namieszane - jak wystawiam nową ofertę to mnie przekierowuje na subdomene staging-lokalnie. Mój login: lukaszw@web-com.pl

**Comment by MaciejFrackowiak** [comment]
@magik092 Czy zakupu dokonałeś z konta firmowego? Jeżeli tak, to jest to prawidłowe zachowanie - anulowanie jest dostępne dla kont zwykłych. Natomiast dla wystawienia oferty skorzystaj proszę z odnośnika https://allegro.pl.allegrosandbox.pl/offer/?lokalnie

**Comment by magik092** [comment]
Dasz radę zmienić typ konta lukaszw na zwykłe konto? Przełącznik "Zmień na konto zwykłe" niestety nie działa lub nie wiem co dalej zrobić aby zmienić typ konta.

**Comment by MaciejFrackowiak** [comment]
Niestety, nie mam takiej możliwości. To konto firmowe, dla którego nie wysłano dokumentów do weryfikacji, ani wcześniej nie zostało aktywowane. Przed momentem je aktywowałem. Opcja zmiany typu konta powinna zadziałać dopiero po jego weryfikacji - w tym celu musiałbyś przesłać dwa dowolne pliki poprzez zakładkę [Dane konta](https://allegro.pl.allegrosandbox.pl/moje-allegro/moje-konto/dane-konta).

**Comment by magik092** [comment]
Jest Pozytywna weryfikacja Allegro, kliknięcie w link Zmień na konto zwykłe przenosi do zakładki Dane osobowe, nie widzę tu nigdzie przełącznika do zmiany typ konta :/
Dobra, szkoda zachodu na to, założyłem nowe konto, to mi wystarczy.

**Comment by pietrach** [comment]
Czy jeśli w panelu Allegro nie ma żadnego abonamentu opłaconego (po stronie sprzedawcy) to klient widzi zmiany statusów swojej oferty ? Np. że w trakcie realizacji, wysłane lub anulowane ?

**Comment by MaciejFrackowiak** [comment]
@pietrach Nie, statusy są niewidoczne dla kupujących. Natomiast powiadomienia o zmianie statusu wysyłane są tylko do kupujących, gdy sprzedający korzysta z abonamentu.

**Comment by bartlomiej-lazarek** [comment]
Konto: IDEAerp_client
Kupiona aukcja: https://allegro.pl.allegrosandbox.pl/oferta/orzechy-6208150587

Nie widzę w menu Kupione -> Szczegóły opcji anulowania tego.
Konto, które sprzedaje jest firmowe. Konto kupującego jest zwykłe.

@MaciejFrackowiak Mógłbyś sprawdzić?

**Comment by MaciejFrackowiak** [comment]
@bartlomiej-lazarek Widzę, że opcja jest już dostępna na koncie kupującego.

**Comment by Onixarts** [comment]
Czy jeśli zamówienie jest już w realizacji, a sprzedający ustawi je z powrotem na nowe to kupującemu pojawi się opcja anulowania czy już nie?

**Comment by MaciejFrackowiak** [comment]
@Onixarts Nie, ponownie opcja anulowania nie będzie dostępna.

**Comment by Onixarts** [comment]
A nie rozważacie aby była? Mamy właśnie taką sytuację u naszego klienta, kupujący chciał zrezygnować jak zamówienie poszło już do realizacji. Wtedy nie da się już skorzystać z szybkiej ścieżki zwrotu prowizji itp.

**Comment by MaciejFrackowiak** [comment]
Dotychczas w kwestii anulowania nie otrzymaliśmy takich sugestii - Twój głos jest pierwszy, przekaże go do osób odpowiedzialnych za rozwój REST API.
